src/Features/CSharp/Portable/UseExpressionBodyForLambda/UseExpressionBodyForLambdaCodeRefactoringProvider.cs

```diff
@@ -0,0 +1,87 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Immutable;
+using System.Composition;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeRefactorings;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Formatting;
+using Microsoft.CodeAnalysis.Options;
+
+namespace Microsoft.CodeAnalysis.CSharp.UseExpressionBodyForLambda
+{
+    using static UseExpressionBodyForLambdaHelpers;
+
+    [ExportCodeRefactoringProvider(LanguageNames.CSharp,
+        Name = PredefinedCodeRefactoringProviderNames.UseExpressionBody), Shared]
+    internal class UseExpressionBodyForLambdaCodeRefactoringProvider : CodeRefactoringProvider
+    {
+        public UseExpressionBodyForLambdaCodeRefactoringProvider()
+        {
+        }
+
+        public override async Task ComputeRefactoringsAsync(CodeRefactoringContext context)
+        {
+            if (context.Span.Length > 0)
+            {
+                return;
+            }
+
+            var position = context.Span.Start;
+            var document = context.Document;
+            var cancellationToken = context.CancellationToken;
+
+            var root = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+            var lambdaNode = root.FindToken(position).Parent.FirstAncestorOrSelf<LambdaExpressionSyntax>();
+            if (lambdaNode == null)
```

Comment:
>lambdaNode [](start = 16, length = 10)

Should the refactoring trigger anywhere inside the lambda, or only on the `=>`? I'd think the latter.

Reply:
We allow it anywhere inside. this is the same behavior as hte normal feature for methods etc.

Note that thsi will only show up for effectively small lambdas.  i.e. ones that actually can go from block to expr, or ones that are only an expr.  So its' generally not a large span this is available in.